crates/ra_hir_def/src/import_map.rs

```diff
@@ -0,0 +1,323 @@
+//! A map of all publicly exported items in a crate.
+
+use crate::{
+    db::DefDatabase,
+    item_scope::ItemInNs,
+    path::{ModPath, PathKind},
+    visibility::Visibility,
+    ModuleDefId, ModuleId,
+};
+use ra_db::CrateId;
+use rustc_hash::FxHashMap;
+use std::{collections::hash_map::Entry, sync::Arc};
```

Comment:
https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/README.md#order-of-imports

Reply:
(not that we follow this strictly... some day, I'd just write assist to do this thing (said @matklad for the nth time))